rustler/src/wrapper/list.rs

```diff
@@ -26,6 +26,23 @@ pub unsafe fn make_list(env: NIF_ENV, arr: &[NIF_TERM]) -> NIF_TERM {
     rustler_sys::enif_make_list_from_array(env, arr.as_ptr(), arr.len() as u32)
 }
 
+pub unsafe fn make_list_from_end(
+    env: NIF_ENV,
+    iter: &mut dyn std::iter::DoubleEndedIterator<Item = NIF_TERM>,
+) -> NIF_TERM {
+    let acc = make_list(env, &[]);
+    iter.rfold(acc, |acc, term| make_list_cell(env, term, acc))
+}
+
+/// Builds a list from an iterator.
+/// Prefer to use make_list_from_end if the iterator is double ended.
```

Comment:
```suggestion
/// Builds a list from an iterator.
///
/// This allocates the iterator as a vector and then builds a list form it.
/// You may also use make_list_from_end if the iterator is double ended,
/// which avoids the intermediate vector allocation but it is slightly slower
/// as it allocates each cons cell one at a time instead of upfront.
```